lib/sudo-exec/src/pty.rs

```diff
@@ -14,32 +19,42 @@ pub(super) struct PtyRelay {
     // FIXME: Look for `SFD_LEADER` occurences in `exec_pty` to decide what to do with the leader
     // side of the pty. It should be used to handle signals like `SIGWINCH` and `SIGCONT`.
     _pty_leader: OwnedFd,
-    rx: OwnedFd,
+    backchannel: ParentBackchannel,
 }
 
 impl PtyRelay {
     pub(super) fn new(
         monitor_pid: ProcessId,
         sudo_pid: ProcessId,
         pty_leader: OwnedFd,
-        rx: OwnedFd,
+        mut backchannel: ParentBackchannel,
     ) -> io::Result<Self> {
+        retry_while_interrupted(|| backchannel.send(&MonitorEvent::ExecCommand))?;
+
         Ok(Self {
             signal_handlers: SignalHandlers::new()?,
```

Comment:
Q: should the signal handler override happen before the command process is spawned or after? given that `SignalHandlers::new` comes after the `send(ExecCommand)` "barrier" it's unspecified whether it'll be executed before or after the command spawn.

Reply:
You're right. The signal handler override should happen way before. I think it should even happen before forking so we can receive `SIGCHLD` if the command exits before we are done initializing the parent process and the monitor. I'm going to move this way up.

Reply:
Now I wonder how our signal handlers interact with `fork`. I'm not sure if the handler will still be valid for the `child` or not.